### PR TITLE
[YamlCreate.ps1] Fix #27518

### LINKED ISSUE #27518: [YamlCreate] does not correctly parse Last Version with .validation file present
# Brief description of your issue

[YamlCreate] does not correctly parse Last Version of the manifest when a `.validation` file is present, i.e. it returns the application shortname as the Last Version instead. 

# Steps to reproduce

Run 

```
.\Tools\YamlCreate.ps1
```

Choose "Update Manifest", enter PackageIdentifier, enter Version number. 

```
Select Mode

[1] New Manifest

[2] Update Manifest

[3] New Locale     

[q] Any key to quit

Selection: 2       

[Required] Enter the Package Identifier, in the following format <Publisher shortname.Application shortname>. For example: Microsoft.Excel
PackageIdentifier: PostgreSQL.pgAdmin

[Required] Enter the version. for example: 1.33.7
Version: 5.7
Last Version: pgAdmin
Get-ChildItem: ...\winget-pkgs\Tools\YamlCreate.ps1:113:36
Line |
 113 |  … cript:OldManifests = Get-ChildItem -Path "$AppFolder\..\$LastVersion"
     |                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | Cannot find path '...\winget-pkgs\manifests\p\PostgreSQL\pgAdmin\pgAdmin' because it does not exist.


Could not find required manifests, input a version containing required manifests

```

# Expected behavior

Return the correct "Last Version" and find the correct path, i.e. `Publisher shortname\Application shortname\Last version`

# Actual behavior

"Last Version: " returns the application shortname, not the last version. The path cannot be found subsequently, because it is looking for a folder named `Publisher shortname\Application shortname\Application shortname`. 

This error is "caused" by the `.validation` file in the `Publisher shortname\Application shortname` folder, in this case https://github.com/microsoft/winget-pkgs/blob/master/manifests/p/PostgreSQL/pgAdmin/.validation, created by #25187. I.e. `YamlCreate` needs to ignore `.validation` files. 

# Environment

```none
[winget --info]
Windows Package Manager v1.0.11692
Windows: Windows.Desktop v10.0.19041.1165
Paket: Microsoft.DesktopAppInstaller v1.12.11692.0
```
winget-pkgs repo updated to commit b097b37543c43ac70e4e49d872db4a898c86cc00

### PR DESCRIPTION
**Pull request by vedantmgoyal9**
- [ ] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [ ] Have you validated your manifest locally with `winget validate --manifest <path>`? 
- [ ] Have you tested your manifest locally with `winget install --manifest <path>`?
- [ ] Does your manifest conform to the [1.0 schema](https://github.com/microsoft/winget-cli/blob/master/doc/ManifestSpecv1.0.md)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----
Fixes #27518

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/27522)

### CODE REVIEW

**Comment by vedantmgoyal2009** [comment]
cc @jedieaston

File: Tools/YamlCreate.ps1 [hunk 1]
```diff
@@ -110,7 +110,7 @@ Function Read-PreviousWinGet-Manifest {
             $LastVersion = Split-Path (Split-Path (Get-ChildItem -Path "$AppFolder\..\" -Recurse -Depth 1 -File).FullName ) -Leaf | Sort-Object $ToNatural | Select-Object -Last 1
 
             Write-Host -ForegroundColor 'DarkYellow' -Object "Last Version: $LastVersion"
-            $script:OldManifests = Get-ChildItem -Path "$AppFolder\..\$LastVersion"
+            $script:OldManifests = Get-ChildItem -Path "$AppFolder\..\$LastVersion" -Directory
 
             if (-not ($OldManifests.Name -like "$PackageIdentifier*.yaml")) {
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:115]: If we use the `-Directory` tag, won't this logic no longer work?

Comment (Trenly) [reply Tools/YamlCreate.ps1:115]: @vedantmgoyal2009 I think you need to do something like what was done in Trenly#9 instead of using the `-Directory`

Comment (vedantmgoyal2009) [reply Tools/YamlCreate.ps1]: Well, I simply copied Easton's suggestion and did not use my brain 🧠

File: Tools/YamlCreate.ps1 [hunk 2]
```diff
@@ -107,7 +107,7 @@ Function Read-WinGet-MandatoryInfo {
 Function Read-PreviousWinGet-Manifest {
     Switch ($Option) {
         'Update' {
-            $LastVersion = Split-Path (Split-Path (Get-ChildItem -Path "$AppFolder\..\" -Recurse -Depth 1 -File).FullName ) -Leaf | Sort-Object $ToNatural | Select-Object -Last 1
+            $LastVersion = Split-Path (Split-Path (Get-ChildItem -Path "$AppFolder\..\" -Recurse -Depth 1 -File -Exclude ".validation").FullName ) -Leaf | Sort-Object $ToNatural | Select-Object -Last 1
```
Comment (Trenly) [inline Tools/YamlCreate.ps1:110]:
```suggestion
            $LastVersion = Split-Path (Split-Path (Get-ChildItem -Path "$AppFolder\..\" -Recurse -Depth 1 -File -Filter "*.yaml").FullName ) -Leaf | Sort-Object $ToNatural | Select-Object -Last 1
```

Oop, I made a mistake too; I forgot that `-Exclude` overrides the `-Depth` as we found in [Trenly#11](https://github.com/Trenly/winget-pkgs/pull/11#issuecomment-896984783) . Based on [this commit](https://github.com/Trenly/winget-pkgs/pull/11/commits/696ab0e7a92bd20426c2c65dce35265256acc477), I think this should work now

**Comment by vedantmgoyal2009** [comment]
@Trenly Can you aprrove changes now?

**Review by Trenly — APPROVED** [review]

**Comment by vedantmgoyal2009** [comment]
@OfficialEsco @denelon

**Review by denelon — APPROVED** [review]

**Pull request merged**